functions/Install-DbaSqlWatch.ps1

```diff
@@ -0,0 +1,231 @@
+                    }
+
+                    # copy the file from temp to local cache
+                    Write-Message -Level Verbose "Copying $zipfile to $LocallyCachedZip"
+                    try {
+                        New-Item -Path $LocallyCachedZip -ItemType File -Force | Out-Null
+                        Copy-Item -Path $zipfile -Destination $LocallyCachedZip -Force
+                    } catch {
+                        # should we stop the function if the file copy fails?
+                    }
+                }
+            }
+        } else {
+
+            # $LocalFile was passed, so use it
+            if ($PSCmdlet.ShouldProcess($env:computername, "Copying local file to temp directory")) {
+
+                if ($LocalFile.EndsWith("zip")) {
+                    $LocallyCachedZip = $zipfile
+                    Copy-Item -Path $LocalFile -Destination $LocallyCachedZip -Force
+                } else {
+                    $LocallyCachedZip = (Join-Path -path $tempFolder -childpath "SqlWatch.zip")
+                    Copy-Item -Path $LocalFile -Destination $LocallyCachedZip -Force
+                }
+            }
+        }
+
+        # expand the zip file
+        if ($PSCmdlet.ShouldProcess($env:computername, "Unpacking zipfile")) {
+
+            Write-Message -Level VeryVerbose "Unblocking $LocallyCachedZip"
+            Unblock-File $LocallyCachedZip -ErrorAction SilentlyContinue
+            $LocalCacheFolder = Split-Path $LocallyCachedZip -Parent
+
+            Write-Message -Level Verbose "Extracting $LocallyCachedZip to $LocalCacheFolder"
+            if (Get-Command -ErrorAction SilentlyContinue -Name "Expand-Archive") {
+                try {
+                    Expand-Archive -Path $LocallyCachedZip -DestinationPath $LocalCacheFolder -Force
+                } catch {
+                    Stop-Function -Message "Unable to extract $LocallyCachedZip. Archive may not be valid." -ErrorRecord $_
+                    return
+                }
+            } else {
+                # Keep it backwards compatible
+                $shell = New-Object -ComObject Shell.Application
+                $zipPackage = $shell.NameSpace($LocallyCachedZip)
+                $destinationFolder = $shell.NameSpace($LocalCacheFolder)
+                Get-ChildItem "$LocalCacheFolder\SqlWatch.zip" | Remove-Item
+                $destinationFolder.CopyHere($zipPackage.Items())
+            }
+
+            Write-Message -Level VeryVerbose "Deleting $LocallyCachedZip"
+            Remove-Item -Path $LocallyCachedZip
+        }
+
+    }
+
+
+    process {
+        if (Test-FunctionInterrupt) {
+            return
+        }
+
+        foreach ($instance in $SqlInstance) {
+            if ($PSCmdlet.ShouldProcess($instance, "Installing SqlWatch")) {
+                try {
+                    Write-Message -Level VeryVerbose -Message "Connecting to $instance."
```

Comment:
This can be removed - it's taken care of inside Connect-SqlInstance

Reply:
Write-Message removed.